field_friend/automations/navigation/field_navigation.py

```diff
@@ -135,24 +166,28 @@ async def _drive(self, distance: float) -> None:
             self._state = await self._run_row_completed()
 
     async def _run_approaching_row_start(self) -> State:
+        self.robot_on_field = False
         self.set_start_and_end_points()
+        if self.start_point is None or self.end_point is None:
+            return State.ERROR
+        if isinstance(self.detector, rosys.vision.DetectorSimulation) and not rosys.is_test:
+            self.create_simulation()
+        else:
+            self.plant_provider.clear()
+
         await self.gnss.ROBOT_POSE_LOCATED.emitted(self._max_gnss_waiting_time)
-        # turn towards row start
-        assert self.start_point is not None
-        target_yaw = self.odometer.prediction.direction(self.start_point)
-        await self.turn_in_steps(target_yaw)
-        # drive to row start
-        await self.drive_in_steps(Pose(x=self.start_point.x, y=self.start_point.y, yaw=target_yaw))
-        await self.gnss.ROBOT_POSE_LOCATED.emitted(self._max_gnss_waiting_time)
+        if not self.robot_on_field:
+            # turn towards row start
+            assert self.start_point is not None
+            target_yaw = self.odometer.prediction.direction(self.start_point)
+            await self.turn_in_steps(target_yaw)
+            # drive to row start
+            await self.drive_in_steps(Pose(x=self.start_point.x, y=self.start_point.y, yaw=target_yaw))
+            await self.gnss.ROBOT_POSE_LOCATED.emitted(self._max_gnss_waiting_time)
         # turn to row
         assert self.end_point is not None
         row_yaw = self.start_point.direction(self.end_point)
```

Comment:
We should calculate the yaw of the robots position to the end point in case of starting the automation while the robot is in working area and has a deviation from the row.

```suggestion
        driving_yaw = self.odometer.prediction.direction(self.end_point)
```